arviz/stats/diagnostics.py

```diff
@@ -4,13 +4,104 @@
 import numpy as np
 import pandas as pd
 from scipy.signal import fftconvolve
+from scipy import stats
+
 import xarray as xr
 
+from .stats_utils import make_ufunc as _make_ufunc
 from ..data import convert_to_dataset
 from ..utils import _var_names
 
 
-__all__ = ["effective_sample_size", "rhat", "geweke", "autocorr"]
+__all__ = [
+    "bfmi",
+    "effective_sample_size",
+    "bulk_effective_sample_size",
+    "tail_effective_sample_size",
+    "rhat",
+    "geweke",
+    "autocorr",
+    "mcse_mean",
+    "mcse_sd",
+    "mcse_quantile",
+]
+
+
+def autocorr(x):
+    """Compute autocorrelation using FFT for every lag for the input array.
+
+    See https://en.wikipedia.org/wiki/autocorrelation#Efficient_computation
+
+    Parameters
+    ----------
+    x : Numpy array
+        An array containing MCMC samples
+
+    Returns
+    -------
+    acorr: Numpy array same size as the input array
+    """
+    y = x - x.mean()
+    len_y = len(y)
+    with warnings.catch_warnings():
+        # silence annoying numpy tuple warning in another library
+        # silence hack added in 0.3.3+
+        warnings.simplefilter("ignore")
+        result = fftconvolve(y, y[::-1])
+    acorr = result[len(result) // 2 :]
+    acorr /= np.arange(len_y, 0, -1)
+    with np.errstate(invalid="ignore"):
+        acorr /= acorr[0]
+    return acorr
+
+
+def _autocov(x):
+    """Compute autocovariance estimates for every lag for the input array.
+
+    Parameters
+    ----------
+    x : Numpy array
+        An array containing MCMC samples
+
+    Returns
+    -------
+    acov: Numpy array same size as the input array
+    """
+    acorr = autocorr(x)
+    varx = np.var(x, ddof=1) * (len(x) - 1) / len(x)
```

Comment:
What is the point with this?

    np.var(x, ddof=1) * (len(x) - 1) / len(x)

is same as

    np.var(x, ddof=0)